### PR TITLE
Disable `clippy::unwrap_or_default` lint for the project

### PR DESCRIPTION
**Pull request by pkhry**
### Description 
see  rust-lang/rust-clippy#12643
subxt uses `darling` which uses `unwrap_or_default` in generated code.

### CODE REVIEW

File: macro/src/lib.rs [hunk 1]
```diff
@@ -3,8 +3,7 @@
 // see LICENSE for license details.
 
 //! Subxt macro for generating Substrate runtime interfaces.
-
-extern crate proc_macro;
+#![allow(clippy::manual_unwrap_or_default, missing_docs)]
```
Comment (niklasad1) [inline macro/src/lib.rs:6]: is this `allow` needed?

**Comment by pkhry** [comment]
Yeah, for some reason, workspace lint config is not getting propagated here

File: macro/src/lib.rs (hunk 1 above)
Comment (pkhry) [reply macro/src/lib.rs:6]: Yeah, for some reason workspace lint is not getting propagated here for manual unwrap

Comment (niklasad1) [inline macro/src/lib.rs:6]: can you remove this `missing_docs` and add `missing_docs` on pub fn subxt?

```suggestion
// TODO: The workspace lint is not working properly so it's disabled here for now.
#![allow(clippy::manual_unwrap_or_default)]
```

File: macro/Cargo.toml [hunk 2]
```diff
@@ -27,3 +27,6 @@ syn = { workspace = true }
 quote = { workspace = true }
 subxt-codegen = { workspace = true, features = ["fetch-metadata"] }
 scale-typegen = { workspace = true }
+
+[lints]
+workspace = true
```
Comment (niklasad1) [inline macro/Cargo.toml:32]: Do you mind adding this to all crates in subxt?

**Review by niklasad1 — APPROVED** [review]

File: macro/Cargo.toml (hunk 2 above)
Comment (pkhry) [reply macro/Cargo.toml:32]: added to all of them, except for examples and testing

File: macro/src/lib.rs [hunk 3]
```diff
@@ -3,8 +3,8 @@
 // see LICENSE for license details.
 
 //! Subxt macro for generating Substrate runtime interfaces.
-
-extern crate proc_macro;
+// TODO: The workspace lint is not working properly so it's disabled here for now.
+#![allow(clippy::manual_unwrap_or_default)]
```
Comment (jsdw) [inline macro/src/lib.rs:7]: Do we need this if you also are pulling lints from the workspace?

File: signer/src/eth.rs [hunk 4]
```diff
@@ -206,7 +206,18 @@ impl Display for AccountId20 {
         write!(f, "{}", self.checksum())
     }
 }
-
+/// Verify that some signature for a message was created by the owner of the [`PublicKey`].
```
Comment (jsdw) [inline signer/src/eth.rs:209]: Nit: add newline :)

```suggestion

/// Verify that some signature for a message was created by the owner of the [`PublicKey`].
```

**Review by jsdw — APPROVED** [review]
Looks good; just a couple of wee things to fix before merging :)

File: macro/src/lib.rs (hunk 3 above)
Comment (niklasad1) [reply macro/src/lib.rs:7]: It doesn't work without this.

I got:

```bash
  process didn't exit successfully: `/home/niklasad1/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/bin/clippy-driver /home/niklasad1/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/bin/rustc --crate-name subxt_macro --edition=2021 macro/src/lib.rs --error-format=json --json=diagnostic-rendered-ansi,artifacts,future-incompat --diagnostic-width=273 --crate-type proc-macro --emit=dep-info,metadata -C prefer-dynamic -C embed-bitcode=no -C debuginfo=2 --deny=unused_parens --deny=unused_extern_crates --deny=unused_comparisons --deny=unused_allocation --deny=unconditional_recursion '--allow=clippy::type_complexity' --deny=patterns_in_fns_without_body --deny=path_statements --deny=overflowing_literals --deny=non_shorthand_field_patterns --deny=no_mangle_generic_items --deny=missing_docs '--allow=clippy::manual-unwrap-or-default' --deny=improper_ctypes --deny=bad_style '--deny=clippy::all' -C metadata=43244fc89d4ec22e -C extra-filename=-43244fc89d4ec22e --out-dir /home/niklasad1/Github/subxt/target/debug/deps -C incremental=/home/niklasad1/Github/subxt/target/debug/incremental -L dependency=/home/niklasad1/Github/subxt/target/debug/deps --extern darling=/home/niklasad1/Github/subxt/target/debug/deps/libdarling-8565ccfffaac534e.rmeta --extern codec=/home/niklasad1/Github/subxt/target/debug/deps/libparity_scale_codec-d4175ab4871a818f.rmeta --extern proc_macro_error=/home/niklasad1/Github/subxt/target/debug/deps/libproc_macro_error-3ca2393b60a689cf.rmeta --extern quote=/home/niklasad1/Github/subxt/target/debug/deps/libquote-123956ecf9e3264f.rmeta --extern scale_typegen=/home/niklasad1/Github/subxt/target/debug/deps/libscale_typegen-24a6bdebbb8b8927.rmeta --extern subxt_codegen=/home/niklasad1/Github/subxt/target/debug/deps/libsubxt_codegen-5eec60c5ec0c06a1.rmeta --extern syn=/home/niklasad1/Github/subxt/target/debug/deps/libsyn-c52220c64d6d07d1.rmeta --extern proc_macro -L native=/home/niklasad1/Github/subxt/target/debug/build/ring-fd1d525f9c796c1d/out` (exit status: 1)
```

So, `'--allow=clippy::manual-unwrap-or-default'` is actually passed when compiling the crate but fails anyway...
That's why this TODO exist :(

Comment (pkhry) [reply macro/src/lib.rs:7]: Yup, this lint setting is not getting propagated for some reason

**Pull request merged**